.github/workflows/build-and-test.yml

```diff
@@ -10,14 +10,20 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
+        with:
+          # Necessary for changeset status to work
+          fetch-depth: 0
       - uses: actions/setup-node@v2
         with:
-          node-version: "16.9.1"
-      - uses: pnpm/action-setup@v2.0.1
+          node-version: "20"
+      - uses: pnpm/action-setup@v2
         with:
-          version: 6.15.0
+          version: 8.10.5
+      # Necessary for changeset status to work
+      - run: git checkout 6.0 && git checkout $GITHUB_SHA
```

Comment:
We'll need to change this to `git checkout main` once the branch renaming is finished